hydromt/data_sources/geodataframe_data_source.py

```diff
@@ -0,0 +1,69 @@
+"""Generic DataSource for GeoDataFrames."""
+
+from logging import Logger
+
+import geopandas as gpd
+from pydantic import field_validator
+
+from hydromt.drivers.geodataframe_driver import GeoDataFrameDriver
+from hydromt.drivers.pyogrio_driver import PyogrioDriver
+from hydromt.metadata_resolvers.resolver_plugin import RESOLVERS
+from hydromt.nodata import NoDataStrategy
+
+from .data_source import DataSource
+
+_KNOWN_DRIVERS: dict[str, GeoDataFrameDriver] = {"pyogrio": PyogrioDriver}
```

Comment:
I think a frozen dict would be better here so it doesn't get moddified accidentally

Reply:
This is a placeholder for plugins later, I should add a comment there.